src/Servers/Kestrel/Core/test/HttpRequestHeadersTests.cs

```diff
@@ -860,6 +860,6 @@ private static string ChangeNameCase(string name, int variant)
 
     // Content-Length is numeric not a string, so we exclude it from the string reuse tests
     public static IEnumerable<object[]> KnownRequestHeaders =>
-        RequestHeaders.Where(h => h.Name != "Content-Length").Select(h => new object[] { true, h }).Concat(
-        RequestHeaders.Where(h => h.Name != "Content-Length").Select(h => new object[] { false, h }));
+        RequestHeaders.Where(h => h.Name != "Content-Length").Select(h => new object[] { true, h.Name }).Concat(
+        RequestHeaders.Where(h => h.Name != "Content-Length").Select(h => new object[] { false, h.Name }));
```

Comment:
Pass name to unit tests instead of KnownHeader so VS test UI shows individual tests. Requires all arguments to be primitive.